### DISCUSSION TITLE
A question about timeouts

### DISCUSSION THREAD

**Discussion by iamleniac** [saul/demofile#251] [Questions] [answered]
Is there possible to detect when a team calls a tactical timeout in demos?
By tactical timeout I mean when the team pauses the game to enable coach communication.
I'm making a study about how coachs can impact game results based on data and statistics and this information is very useful. If I can get this just from the demos without the need to watch them it would be great!

**Comment by saul** [comment]
You should be able to listen out for the `svc_SetPause` message. However you won't be able to determine who instigated the pause from that alone. You'd probably be able to determine this by what's been put in the chat, e.g. by parsing chat messages similar to:

https://github.com/saul/demofile/blob/2e1aeac4e5dfe999cad441b1fad567000e3e64da/src/examples/dumpfile.ts#L145-L157

Let me know if this doesn't solve your issue.

**Comment by iamleniac** [comment, marked as answer]
Hi, @saul I've found a way to do what I needed.

I watched a recorded match and compared with my results, and it works!

Let me share my code: 

```javascript
const fs = require("fs")
const demofile = require("demofile")

fs.readFile("natus-vincere-vs-astralis-m1-nuke.dem", (err, buffer) => {
  const demoFile = new demofile.DemoFile()

  demoFile.entities.on("change", e => {
    if (
      e.tableName == "DT_VoteController" &&
      e.varName == "m_iOnlyTeamToVote" &&
      e.newValue !== -1
    ) {
      demoFile.teams.map(team => {
        if (
          team.props.DT_Team.m_iTeamNum ==
          e.entity.props.DT_VoteController.m_iOnlyTeamToVote
        ) {
          console.log(
            `Team called timeout: ${team.props.DT_Team.m_szClanTeamname}`
          )
        }
      })
    }
  })

  demoFile.parse(buffer)
})
```

In this particular match, at the Faceit Major grand finals, Astralis called 3 pauses and Navi called 2.

Is it possible to get the round number this event occurred?

I can open another issue on that if you want.

Thanks for everything so far.

**Comment by saul** [comment]
Yes - you can use `DemoFile#gameRules.roundsPlayed`. See the docs: https://saul.github.io/demofile/classes/_entities_gamerules_.gamerules.html#roundsplayed
